webpack/ForemanTasks/Components/common/ActionButton/ActionButton.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { Button } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+export const CancelButton = ({ id, name, cancellable, cancel }) => (
+  <Button
+    bsSize="small"
+    onClick={() => cancel(id, name)}
+    disabled={!cancellable}
+  >
+    {__('Cancel')}
+  </Button>
+);
+
+export const ResumeButton = ({ id, name, resumable, resume }) => (
+  <Button bsSize="small" onClick={() => resume(id, name)} disabled={!resumable}>
+    {__('Resume')}
+  </Button>
+);
+
+export const ActionButton = ({ id, name, availableActions, taskActions }) => {
+  const isResume = availableActions.resumable;
+  if (isResume) {
+    return (
+      <ResumeButton id={id} name={name} resume={taskActions.resume} resumable />
+    );
+  }
+  return (
+    <CancelButton
+      id={id}
+      name={name}
+      cancellable={availableActions.cancellable}
+      cancel={taskActions.cancel}
+    />
+  );
+};
+
+ResumeButton.propTypes = {
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+  resume: PropTypes.func.isRequired,
+  resumable: PropTypes.bool.isRequired,
```

Comment:
Shouldn't we use default values for the buttons?
```js
ResumeButton.defaultProps = {
  resume: noop,
  resumable: false,
};
```

Reply:
Why?

Reply:
So it becomes more generic

Reply:
Why would anyone need a button that just says `resume` with no functionality?

Reply:
I don't believe someone will use it without functionality but it will reduce the gap needed to render the component and experience with it.

We do have a general rule that says if I can provide a default value without breaking the component, it is better to do so.

Reply:
why a "button without a functionality" is not a broken component?

Reply:
Because I can still render it and experience with it